backend/src/models/reports/project.js

```diff
@@ -0,0 +1,197 @@
+    .rightJoin(`${healthIndicatorTable} as hi`, { "hi.id": "pd.health_id" })
+    .where((builder) => {
+      builder.whereNull("pd.is_expense").orWhere("pd.is_expense", "False");
+    })
+    .andWhere({ "p.id": projectId });
+};
+
+/**
+ * Get the milestones for a specific project by id.
+ *
+ * @param   {int}   projectId The project id.
+ * @returns {any[]}
+ */
+const getMilestones = (projectId) => {
+  return knex(projectMilestoneTable)
+    .select(
+      "project_id",
+      "description",
+      "fiscal_id",
+      {
+        target_completion_date: knex.raw(
+          `TO_CHAR(target_completion_date :: DATE, '${dateFormat}')`
+        ),
+      },
+      {
+        actual_completion_date: knex.raw(
+          `TO_CHAR(actual_completion_date :: DATE, '${dateFormat}')`
+        ),
+      },
+      "status",
+      "health_id"
+    )
+    .where({ project_id: projectId });
+};
+
+/**
+ * Get the strategic alignment for a specific project by id.
+ *
+ * @param   {int}   projectId The project id.
+ * @returns {any[]}
+ */
+const getStrategicAlignment = (projectId) => {
+  return knex(projectStrategicAlignmentTable)
+    .select("strategic_alignment.description")
+    .leftJoin(strategicAlignmentTable, { strategic_alignment_id: "strategic_alignment.id" })
+    .where({ project_id: projectId })
+    .andWhere({ checked: true });
+};
+
+/**
+ * Individual Project Reports - Project Status Summary
+ *
+ * @param   {int}   projectId The project id.
+ * @returns {any[]}
+ */
+const getProjectStatuses = (projectId) => {
+  return knex(`${projectStatusTable} as ps`)
+    .select(
+      "ps.*",
+      { reported_by: knex.raw("c.last_name || ', ' || c.first_name") },
+      { phase: "pp.phase_name" },
+      {
+        project_health: "health.health_name",
+      },
+      {
+        schedule_health: "schedule.health_name",
+      },
+      {
+        budget_health: "budget.health_name",
+      },
+      {
+        team_health: "team.health_name",
+      }
+    )
+    .join(`${contactTable} as c`, "ps.reported_by_contact_id", "c.id")
+    .join(`${projectPhaseTable} as pp`, "ps.project_phase_id", "pp.id")
+    .leftJoin(`${healthTable} as health`, "ps.health_id", "health.id")
+    .leftJoin(`${healthTable} as schedule`, "ps.schedule_health_id", "schedule.id")
+    .leftJoin(`${healthTable} as budget`, "ps.budget_health_id", "budget.id")
+    .leftJoin(`${healthTable} as team`, "ps.team_health_id", "team.id")
+    .where("ps.project_id", projectId)
+    .orderBy("ps.id", "DESC");
+};
+
+/**
+ * Get the lessons learned for a specific project by id.
+ *
+ * @param   {int}   projectId The Project Id
+ * @returns {any[]}
+ */
+const getLessonsLearned = (projectId) => {
+  return knex(lessonsLearnedTable).select("*").where("project_id", projectId);
+};
+
+module.exports = {
+  getProjectById,
+  projectStatusReport,
+  getMilestones,
+  getStrategicAlignment,
+  getProjectStatuses,
+  getLessonsLearned,
+};
```

Comment:
nice idea, breaking out the project related models to a separate  file

Reply:
looks good but I think we can break this out a bit more. Ill have a think on it.